packages/m2m-gateway/test/integration/agreements/createAgreement.test.ts

```diff
@@ -0,0 +1,122 @@
+import { describe, it, expect, vi, beforeEach } from "vitest";
+import { agreementApi, m2mGatewayApi } from "pagopa-interop-api-clients";
+import { generateMock } from "@anatine/zod-mock";
+import {
+  agreementService,
+  expectApiClientPostToHaveBeenCalledWith,
+  mockInteropBeClients,
+  mockPollingResponse,
+} from "../../integrationUtils.js";
+import { PagoPAInteropBeClients } from "../../../src/clients/clientsProvider.js";
+import { config } from "../../../src/config/config.js";
+import {
+  missingMetadata,
+  resourcePollingTimeout,
+} from "../../../src/model/errors.js";
+import {
+  getMockM2MAdminAppContext,
+  getMockedApiAgreement,
+} from "../../mockUtils.js";
+
+describe("createAgreement", () => {
+  const mockAgreementSeed: m2mGatewayApi.AgreementSeed = generateMock(
+    m2mGatewayApi.AgreementSeed
+  );
+
+  const mockAgreementProcessResponse = getMockedApiAgreement({
+    state: agreementApi.AgreementState.Values.DRAFT,
+    eserviceId: mockAgreementSeed.eserviceId,
+    descriptorId: mockAgreementSeed.descriptorId,
+  });
+
+  const mockCreateAgreement = vi
+    .fn()
+    .mockResolvedValue(mockAgreementProcessResponse);
+
+  const mockGetAgreement = vi.fn(
+    mockPollingResponse(mockAgreementProcessResponse, 2)
+  );
+
+  mockInteropBeClients.agreementProcessClient = {
+    createAgreement: mockCreateAgreement,
+    getAgreementById: mockGetAgreement,
+  } as unknown as PagoPAInteropBeClients["agreementProcessClient"];
+
+  beforeEach(() => {
+    // Clear mock counters and call information before each test
+    mockCreateAgreement.mockClear();
+    mockGetAgreement.mockClear();
+  });
+
+  it("Should succeed and perform API clients calls", async () => {
+    const m2mAgreementResponse: m2mGatewayApi.Agreement = {
+      id: mockAgreementProcessResponse.data.id,
+      eserviceId: mockAgreementProcessResponse.data.eserviceId,
+      descriptorId: mockAgreementProcessResponse.data.descriptorId,
+      producerId: mockAgreementProcessResponse.data.producerId,
+      consumerId: mockAgreementProcessResponse.data.consumerId,
+      state: mockAgreementProcessResponse.data.state,
+      createdAt: mockAgreementProcessResponse.data.createdAt,
+    };
+
+    const result = await agreementService.createAgreement(
+      mockAgreementSeed,
+      getMockM2MAdminAppContext()
+    );
+
+    expect(result).toEqual(m2mAgreementResponse);
+    expectApiClientPostToHaveBeenCalledWith({
+      mockPost: mockInteropBeClients.agreementProcessClient.createAgreement,
+      body: mockAgreementSeed,
+    });
+  });
```

Comment:
Also add expectApiClientGetToHaveBeenCalledWith for the polling get, and also the expectation on the number of times it was called (toHaveBeenCalledTimes(2))